Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -3241,8 +3223,10 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
     }
     if let binaryOperator = operatorExpr.as(BinaryOperatorExprSyntax.self) {
       let operatorText = binaryOperator.operatorToken.text
-      if let precedence = operatorContext.infixOperator(named: operatorText)?.precedenceGroup,
-        precedence === operatorContext.precedenceGroup(named: .assignment)
+      // TODO: `OperatorTable` doesn't give us a way to look up the precedence group of an
+      // operator, so we have to hard-code some operators instead of relying on the precedence
+      // groups as we did before. Look at adding this API to `OperatorTable`.
+      if operatorText.hasSuffix("=") && !nonAssignmentOperatorsEndingInEqual.contains(operatorText)
```

Comment:
@DougGregor Any opposition to adding a precedence group lookup function to `OperatorTable` so that we can move back to the more general way of writing this (similarly on lines 3297 and 3348)?

Reply:
I think adding a precedence group lookup function is a great idea. Do you want to come up with a PR against swift-syntax to add the API you like?

Reply:
Already on it 🙂

Reply:
Thanks for the quick merge! I've switched back to the precedence-group-based behavior in this PR. Aside from that, `SwiftOperators` is doing everything I need without any other changes. 👍🏻